workspaces/diff-engine/tests/lib/spawn.js

```diff
@@ -0,0 +1,102 @@
+const Tap = require('tap');
+const fs = require('fs');
+const path = require('path');
+const readline = require('readline');
+const { DiffEngineError } = require('../../lib');
+
+Tap.test('diff-engine.spawn', async (test) => {
+  await test.test('can diff a stream of interactions', async (t) => {
+    const interactions = fs.createReadStream(
+      path.join(
+        __dirname,
+        '..',
+        'fixtures',
+        'ergast-captures',
+        'ergast-simulated-traffic.jsonl'
+      )
+    );
+
+    const DiffEngine = require('../../lib');
+
+    const diff = DiffEngine.spawn({
+      specPath: path.join(
+        __dirname,
+        '..',
+        'fixtures',
+        'ergast-example-spec.json'
+      ),
+    });
+
+    interactions.pipe(diff.input);
+    let outputLines = bufferLines(diff.output);
+    let errorLines = bufferLines(diff.error);
+
+    let diffs = (await outputLines).map((line) => JSON.parse(line));
+    diffs.sort(([, tagsA, fingerprintA], [, tagsB, fingerprintB]) =>
+      compareString(
+        fingerprintA + tagsA.join(','),
+        fingerprintB + tagsB.join(',')
+      )
+    );
+
+    t.matchSnapshot(diffs, 'generated diffs');
+  });
+
+  await test.test(
+    'will reject result with errors for diff-engine process faults',
+    async (t) => {
+      const interactions = fs.createReadStream(
+        path.join(
+          __dirname,
+          '..',
+          'fixtures',
+          'ergast-captures',
+          'ergast-simulated-traffic.jsonl'
+        )
+      );
+
+      const DiffEngine = require('../../lib');
+
+      const diff = DiffEngine.spawn({
+        specPath: path.join(__dirname, 'spawn.js'), // not a spec file
+      });
+
+      interactions.pipe(diff.input);
+      let outputLines = bufferLines(diff.output);
+      let errorLines = bufferLines(diff.error);
+
+      let engineError;
+      try {
+        await diff.result;
+      } catch (err) {
+        engineError = err;
+      }
+
+      t.type(engineError, DiffEngineError);
```

Comment:
if it's not too much trouble, can we snapshot the outputLines and errorLines for each test?

Reply:
Tried, but too much trouble. I considered the same benefits, but already in development of tests ran into the noise and trouble that creates, as both make no guarantee on order of output, hence the parsing and sorting before. In addition to that, stderr yields debug output that's volatile (especially since we don't have a proper logging solution yet). 

In the end, I reckoned these tests are mostly about verifying the integration between the Node.js and Rust side, not exactly what output the diff engine produces. Of course, the two are related, but given the trouble of doing much more than what we're doing now, I found this to be a reasonable balance.